x/finality/abci.go

```diff
@@ -20,6 +20,11 @@ func BeginBlocker(ctx context.Context, k keeper.Keeper) error {
 func EndBlocker(ctx context.Context, k keeper.Keeper) ([]abci.ValidatorUpdate, error) {
 	defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), telemetry.MetricKeyEndBlocker)
 
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	if uint64(sdkCtx.HeaderInfo().Height) < k.GetParams(ctx).FinalityActivationHeight {
+		return []abci.ValidatorUpdate{}, nil
+	}
+
 	// if the BTC staking protocol is activated, i.e., there exists a height where a finality provider
 	// has voting power, start indexing and tallying blocks
 	if _, err := k.BTCStakingKeeper.GetBTCStakingActivatedHeight(ctx); err == nil {
```

Comment:
nit: I think we can combine the two checks into one validation like `validateActivation` indication that we only handle finality when (1) the height higher than the activation height and (2) at least one fp has voting power at this height